AzureMonitorAgent/ama_tst/modules/install/install.py

```diff
@@ -0,0 +1,97 @@
+import os
+import subprocess
+
+from error_codes  import *
+from errors       import error_info, is_error, print_errors
+from .check_os    import check_os
+from .check_pkgs  import check_packages, check_syslog
+from .check_ama   import check_ama
+from helpers      import find_package_manager
+
+# check space in MB for each main directory
+def check_space():
+    dirnames = ["/etc", "/opt", "/var"]
+    for dirname in dirnames:
+        space = os.statvfs(dirname)
+        free_space = space.f_bavail * space.f_frsize / 1024 / 1024
+        if (free_space < 500):
+            error_info.append((dirname, free_space))
+            return ERR_FREE_SPACE
+    return NO_ERROR
+
+
+def check_pkg_manager():
+    pkg_manager = find_package_manager()
+    if (pkg_manager == ""):
+        return ERR_PKG_MANAGER
+    return NO_ERROR
+
+def check_syslog_user():
+    with open('/etc/passwd', 'r') as fp:
+        for line in fp:
+            if line.startswith('syslog:'):
+                return NO_ERROR
+    return ERR_SYSLOG_USER
+
+# check all packages are installed
+def check_installation(interactive, err_codes=True, prev_success=NO_ERROR):
```

Comment:
thought for the future: python is also an AMA dependency. obviously if we are running the troubleshooter already, there is python on the machine since TST is written in python. might be worth having TST entrypoint be a shell script shim (like shim.sh calling agent.py) which makes sure there is python and lets the user know if its not found.

Reply:
The entry point of AMA TST is ama_troubleshooter.sh. The script file checks if python is installed and displays error if python is not found

Reply:
My mistake, thanks for clarifying